Scrape Artists from Musicbrainz

metavaults opened the pull request.

metavaults committed:
Scrape Artists from Musicbrainz

sacdallago commented:
Put someone as reviewer(s) (not us mentors! We are just reviewing these first PRs thoroughly to teach you _how it's done_, next times we will limit ourselves to close PRs if it's not alright <sorry in advance, happens.>)

sacdallago reviewed (dismissed):
No scraped data on the repo online! `.gitignore` it and remove it from the PR

kordianbruck commented on scrapeArtists/server.js:
```diff
@@ -0,0 +1,72 @@
+            request({
+                url: url,
+                headers: {
+                'User-Agent': 'ClassicalApp/1.4.0 ( tumiss@gmail.com )'
```
You should be careful to commit an email address. Github is crawled for such sensitive information

sacdallago commented:
Updates on this?

TimHenkelmann committed:
- Removed package.json and outputfile artistsFromRecordings.json
- added .gitignore and artistsFromRecordings_example.json
- applied changes to server.js:
    - removed server
    - removed email

TimHenkelmann commented:
Applied tutor suggestions for improvement:
- Removed package.json and outputfile artistsFromRecordings.json
- added .gitignore and artistsFromRecordings_example.json
- applied changes to server.js:
    - removed server
    - removed email

Ready for review!

TimHenkelmann commented:
Hi @sacdallago. I have a question for this one. I removed the scraped output (artistsFromRecordings.js) from the PR. However, the server.js script needs to read data from "IDArtistsRecordingsFinal2.json" (an input file which contains scraped data from another script).
How should we proceed with that:
- Remove the input file as well
- Rename it and only show a few entries (then one would also need to change the server.js file)
- Keep the name of the input file and only have it contain a few entries
-  Something completely different

Thanks in advance

kordianbruck commented on scrapeArtists/artistsFromRecordings_example.json:
```diff
@@ -0,0 +1 @@
+{"table":[{"artistId":"e8c91353-e251-456b-80ad-d10c2cd00be3","aboutartist":{"gender":null,"isnis":[],"id":"e8c91353-e251-456b-80ad-d10c2cd00be3","life-span":{"ended":false,"end":null,"begin":null},"name":"University of Arkansas Chorus","begin_area":null,"type":"Choir","sort-name":"University of Arkansas Chorus","country":null,"area":{"id":"8788d6c2-c779-4be5-ad47-cf0a95e0f2a0","disambiguation":"" ...
```
Please don't commit these json files. It only blows up your Git History and makes it slow.

kordianbruck replied:
Or is it intended?

TimHenkelmann replied:
@kordianbruck Yes, this is only an extract of the complete json output file and displays how the data is going to look like

kordianbruck commented:
@TimHenkelmann - you should only write files to disk as a cache. Ideally we only start one server / script that will handle all scraping and loading while keeping everything in ram. (We have kinda unlimited Ram)

sacdallago commented:
Mh, if I understood correctly: this script has a dependency on another script to get data, right?

Natural thing to do would chain these two scripts together, something like:

```
IF NOT dataFromScriptOne exists:
   require scriptOne
   run scriptOne.getData
   run script
ELSE
   run script
```

metavaults committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into LukasNavickas-patch-2

metavaults committed:
fixed according to reviews

- created if/else statements to check if data is available
- put scripts in separate functions
- removed writing to a file (saved to a variable instead)

metavaults reviewed (commented):
- did all changes according to reviewers

metavaults committed:
Revert "fixed according to reviews"

This reverts commit 41f0d1d218e94f8c24c4412dd4e34d6ee192db3b.

kordianbruck commented:
@MusicConnectionMachine/group-1 please also review this! Its been waiting for **7 days** - ahhhh :angry:

metavaults committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into LukasNavickas-patch-2

metavaults committed:
remove json files and fix issues with the app

TimHenkelmann commented:
Hi @kordianbruck , there are still "requested changes" from your side (probably from the .json file that we had in there at first but now deleted). Could you accept them or comment on that?
Thx!

kordianbruck commented on scrapeArtists/server.js:
```diff
@@ -0,0 +1,86 @@
+      return artistai.artistId;
+    });
+
+    setInterval(function(){      
```
Why are you using a setInterval here?

metavaults replied:
In order to get 100% of results, we cannot query MusicBrainz faster than every second. 
Reference: https://musicbrainz.org/doc/XML_Web_Service/Rate_Limiting

kordianbruck replied:
Ugly solution. Might want to reconsider this in the future.

TimHenkelmann replied:
Alright, we will try to find other approaches for this in the future!

kordianbruck replied:
Yea, don't worry about it now.

kordianbruck reviewed (changes requested):
Last thing.
